### PR TITLE
feat: imported vars as override values

### PR DESCRIPTION
**Pull request by decleaver**
## Description

This pr allows you to use imported variables as override values in your uds-bundle.yaml

## Related Issue

Relates to #370 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Other (security config, docs update, etc)

## Checklist before merging

- [X] Test, docs, adr added or updated as needed
- [ ] [Contributor Guide Steps](https://github.com/defenseunicorns/uds-cli/blob/main/CONTRIBUTING.md)(https://github.com/defenseunicorns/uds-template-capability/blob/main/CONTRIBUTING.md#submitting-a-pull-request) followed

### CODE REVIEW

File: src/pkg/bundle/deploy.go [hunk 1]
```diff
@@ -409,6 +410,21 @@ func addOverrideValue(overrides map[string]map[string]*values.Options, component
 		val := fmt.Sprintf("%s=%s", valuePath, j)
 		overrides[component][chart].JSONValues = append(overrides[component][chart].JSONValues, val)
 	default:
+		// Check for any templated variables if pkgVars set
```
Comment (naveensrinivasan) [inline src/pkg/bundle/deploy.go:413]: 1. Optimize Regular Expression Compilation:
Currently, the regular expression is compiled within the executed block of code. This is efficient for occasional use, but if this function is called frequently, it might lead to unnecessary overhead. A common practice is to compile regular expressions once and reuse them throughout the application. This can be achieved by declaring the compiled regex outside of the function scope. This change could improve the performance of your code.

```go
// Consider moving this outside the function to compile it only once
   var templatedVarRegex = regexp.MustCompile(`\${([^}]+)}`)
 ```
 
 2. Enhance Handling of Multiple Templated Variables:
The current implementation replaces the first occurrence of a templated variable. If a string contains multiple templated variables, it would be beneficial to replace all of them. This adjustment ensures that all templated variables are processed, making the function more versatile.

```go
   // Use ReplaceAllStringFunc to handle all occurrences of templated variables
   replacedValue := templatedVarRegex.ReplaceAllStringFunc(templatedVariable, func(match string) string {
       variableName := templatedVarRegex.FindStringSubmatch(match)[1]
       if varValue, ok := pkgVars[variableName]; ok {
           return varValue
       }
       // Consider handling missing variables more explicitly
       return fmt.Sprintf("${%s_not_found}", variableName)
   })
   value = replacedValue
  ```

File: docs/overrides.md [hunk 2]
```diff
@@ -130,6 +130,7 @@ The `value` is the value to set at the `path`. Values can be simple values such
               value:
                 customAnnotation: "customValue"
 ```
+If [importing](../README.md#importingexporting-variables) a variable from another package, that variable can also be used to set a value, using the template syntax `${...}`
```
Comment (UncleGedd) [inline docs/overrides.md:133]: Can we add an explicit example of what this looks like?

File: src/test/bundles/12-exported-pkg-vars/uds-bundle.yaml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        package: output-var
+
+    overrides:
+      podinfo-component:
+        unicorn-podinfo:
+          values:
+            - path: "podinfo.replicaCount"
+              value: 1
+            - path: "podinfo.ui.color"
+              value: ${COLOR}
```
Comment (UncleGedd) [inline src/test/bundles/12-exported-pkg-vars/uds-bundle.yaml:28]: Can we also test the edge case of `${FOO} - ${BAR}` <- or something to that effect

Comment (mjnagel) [inline src/test/bundles/12-exported-pkg-vars/uds-bundle.yaml:28]: Is there a reason this wouldn't be a `variable` with name `color` instead of `value`? This definitely works but feels a bit weird to me since I would typically think of `values` being "hardcoded" in a bundle and `variables` being dynamic.

Comment (UncleGedd) [reply src/test/bundles/12-exported-pkg-vars/uds-bundle.yaml:28]: Good catch @mjnagel ! The vars are set at deploy time so the template probably doesn't make sense under `values`, only under `variables`

Comment (decleaver) [reply src/test/bundles/12-exported-pkg-vars/uds-bundle.yaml:28]: I can switch it if that makes more sense, I just went with `values` because under `variables` it would get set under `default` since there is no value field for variables.

Comment (UncleGedd) [reply src/test/bundles/12-exported-pkg-vars/uds-bundle.yaml:28]: Good point @decleaver , looks like the issue is to have this under `values` 🤔

Comment (mjnagel) [reply src/test/bundles/12-exported-pkg-vars/uds-bundle.yaml:28]: After our synchronous chat I'm on board with `values`. As @UncleGedd explained it `variables` is more for things that the bundle deployer can configure. This is more in line with `values` that aren't configurable by the bundle deployer.

Comment (decleaver) [reply src/test/bundles/12-exported-pkg-vars/uds-bundle.yaml:28]: After discussing, going to stick with setting in `values` vs `variables`

File: docs/overrides.md (hunk 2 above)
Comment (decleaver) [reply docs/overrides.md]: added in latest commit.

File: src/test/bundles/12-exported-pkg-vars/uds-bundle.yaml (hunk 3 above)
Comment (decleaver) [reply src/test/bundles/12-exported-pkg-vars/uds-bundle.yaml:37]: added test case for this, and additional test cases for checking object and list override values.

File: docs/overrides.md [hunk 4]
```diff
@@ -130,6 +130,37 @@ The `value` is the value to set at the `path`. Values can be simple values such
               value:
                 customAnnotation: "customValue"
 ```
+If [importing](../README.md#importingexporting-variables) a variable from another package, that variable can also be used to set a value, using the template syntax `${...}`. In the example below the `COLOR` variable is being used to set the `podinfo.ui.color` value.
```
Comment (UncleGedd) [inline docs/overrides.md:133]:
```suggestion
If using a variable that has been [exported](../README.md#importingexporting-variables) from another package, that variable can also be used to set a value, using the syntax `${...}`. In the example below the `COLOR` variable is being used to set the `podinfo.ui.color` value.
```

File: docs/overrides.md [hunk 5]
```diff
@@ -130,6 +130,37 @@ The `value` is the value to set at the `path`. Values can be simple values such
...
+    ref: 0.0.1
+    exports:
+      - name: COLOR
+
+  - name: helm-overrides
+    path: "../../packages/helm"
+    ref: 0.0.1
+    imports:
+      - name: COLOR
+        package: output-var
```
Comment (UncleGedd) [inline docs/overrides.md:153]:
```suggestion
```

**Review by UncleGedd — APPROVED** [review]
🚀
